plugin-src/gitVisualization/gitWitchcraft.ts

```diff
@@ -34,11 +48,39 @@ export async function visualizeChangedFiles(changedFiles: ChangedFile[], isZoomi
 
 async function zoomToChanges(absoluteFilePaths: string[]): Promise<void> {
     const rootFolder: RootFolderBox = getRootFolder()
-    const renderedBoxes: Box[] = absoluteFilePaths.map(path => {
-        if (existsSync(path)) {
-            return rootFolder.getRenderedBoxesInPath(path).at(-1)
-        }
+    const pathsOfExistingFiles = absoluteFilePaths.filter(async path =>
+        await fileSystem.doesDirentExist(path))
+    const renderedBoxes: Box[] = pathsOfExistingFiles.map(path => {
+        return rootFolder.getRenderedBoxesInPath(path).at(-1)
     }).filter(box => box) as Box[]
     const map: Map = getMapOrError()
     await map.zoomToFitBoxes(renderedBoxes)
+}
+
+export async function openChanges(changedFile: ChangedFile): Promise<void> {
+    let refsForGitDiffTool: string = selectedRefs.at(0)!
+    if (selectedRefs.length > 1) {
+        refsForGitDiffTool = `"${refsForGitDiffTool}" "${selectedRefs.at(-1)}^"`
+    } else if (refsForGitDiffTool !== 'HEAD') {
+        refsForGitDiffTool = `"${refsForGitDiffTool}" "${refsForGitDiffTool}^"`
+    }
+
+    const cwd: string = getRootFolder().getSrcPath()
+    configureGitDifftoolIfNotSet(cwd)
```

Comment:
this is smart 👍 , but keep in mind that there's a race condition. It could happen that `process.on` in `configureGitDifftoolIfNotSet(cwd)` is fired when `environment.runShellCommand(gitDiffCommand, {cwd})` is already called